sirepo/uri_router.py

```diff
@@ -157,11 +167,14 @@ def register_api_module(module=None):
     if hasattr(m, 'init_apis'):
         m.init_apis()
     # It's ok if there are no APIs
-    for n, o in inspect.getmembers(m):
-        if n.startswith(_FUNC_PREFIX) and inspect.isfunction(o):
+    if not hasattr(m, 'API'):
```

Comment:
Document why this test exists. It doesn't seem like the right place. Probably should be in `_register_sim_api_module`.

Reply:
@robnagler Without this check, tests fail because some modules like sirepo.auth.basic don't have any APIs and therefore don't have the class. Could you expand on why `_register_sim_api_modules` is the correct place and how exactly to add it there?

Reply:
Ah, I didn't remember that. I was thinking it was only sim api modules that didn't have APIs. Leave as is, then, and document including an example module.

Technically, this is an abuse of uri_router to enforce initialization. We really should sirepo.init (or something) that modules register with and declare dependencies. I thought [about this 7 years ago](https://github.com/radiasoft/pykern/wiki/ImportThoughts) (almost to the day), but never implemented. The ad hoc initialization that we are doing now is not working, because it requires too much special knowledge/handling. I'll add an issue.

Reply:
3ccfbc6e956420f7da6fc23176cdebb7ae57cfc9